mfr/providers/osf/provider.py

```diff
@@ -66,23 +66,24 @@ def __init__(self, request, url):
             metadata_url = download_url.replace('/file?', '/data?', 1)
             metadata_request = await self._make_request('GET', metadata_url)
             metadata = await metadata_request.json()
+            await metadata_request.release()
         else:
             # URL is for WaterButler v1 API
             self.metrics.add('metadata.wb_api', 'v1')
             metadata_request = await self._make_request('HEAD', download_url)
-            # To make changes to current code as minimal as possible
-            try:
-                metadata = {'data': json.loads(metadata_request.headers['x-waterbutler-metadata'])['attributes']}
-                await metadata_request.release()
-            except KeyError:
-                raise exceptions.MetadataError(
-                    'Failed to fetch metadata. Received response '
-                    'code {}'.format(str(metadata_request.status)),
+            request_code = metadata_request.status
+            request_reason = metadata_request.reason
+            request_headers = metadata_request.headers
+            await metadata_request.release()
+            if request_code != 200:
+                raise exceptions.MetadataError('Failed to fetch file metadata from WaterButler. Received response: code {} {}'.format(str(request_code), str(request_reason)),
                     metadata_url=download_url,
-                    response=await metadata_request.text(),
+                    response=request_reason,
                     provider=self.NAME,
-                    code=400,
-                )
+                    code=400)
```

Comment:
One for `200` and one for `400`. It will be great if we can gradually replace number literals with status constants in our code base ([link](https://docs.python.org/3.5/library/http.html#http.HTTPStatus)). For example:
```python
from http import HTTPStatus

httpStatus.OK
httpStatus.BAD_REQUEST
```

Reply:
Seems like a good idea.

Reply:
There are too many status in the file. No need to change for now. Maybe in the future we can have a clean ticket for this but very low priority for now.